llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -3098,6 +3098,15 @@ bool isKnownNonZero(const Value *V, const APInt &DemandedElts,
   // Check for pointer simplifications.
 
   if (PointerType *PtrTy = dyn_cast<PointerType>(Ty)) {
+    const DataLayout &DL = Q.DL;
+    if (DL.isSentinelValueDefined() && PtrTy) {
```

Comment:
PtrTy check is redundant

Reply:
> Fold to return of boolean expression
> PtrTy check is redundant

removed PtrTy check and updated fold to return.